cpp/models/lct_secir/model.h

```diff
@@ -35,233 +35,372 @@ namespace mio
-                subcompartments_ts[i]
-                    .segment(LctState::template get_first_index<InfectionState::Exposed>(),
-                             LctState::template get_num_subcompartments<InfectionState::Exposed>())
-                    .sum();
-            compartments[(Eigen::Index)InfectionState::InfectedNoSymptoms] =
-                subcompartments_ts[i]
-                    .segment(LctState::template get_first_index<InfectionState::InfectedNoSymptoms>(),
-                             LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>())
-                    .sum();
-            compartments[(Eigen::Index)InfectionState::InfectedSymptoms] =
-                subcompartments_ts[i]
-                    .segment(LctState::template get_first_index<InfectionState::InfectedSymptoms>(),
-                             LctState::template get_num_subcompartments<InfectionState::InfectedSymptoms>())
-                    .sum();
-            compartments[(Eigen::Index)InfectionState::InfectedSevere] =
-                subcompartments_ts[i]
-                    .segment(LctState::template get_first_index<InfectionState::InfectedSevere>(),
-                             LctState::template get_num_subcompartments<InfectionState::InfectedSevere>())
-                    .sum();
-            compartments[(Eigen::Index)InfectionState::InfectedCritical] =
-                subcompartments_ts[i]
-                    .segment(LctState::template get_first_index<InfectionState::InfectedCritical>(),
-                             LctState::template get_num_subcompartments<InfectionState::InfectedCritical>())
-                    .sum();
-            compartments[(Eigen::Index)InfectionState::Recovered] =
-                subcompartments_ts[i][LctState::template get_first_index<InfectionState::Recovered>()];
-            compartments[(Eigen::Index)InfectionState::Dead] =
-                subcompartments_ts[i][LctState::template get_first_index<InfectionState::Dead>()];
+    }
 
-            compartments_ts.add_time_point(subcompartments_ts.get_time(i), compartments);
+    /**
+     * @brief Checks whether LctState of a group satisfies all constraints. 
+     *  Recursively, it checks that all groups satisfy the constraints.
+     *
+     * @tparam group The group for which the constraints should be checked.
+     * @return Returns true if one or more constraints are not satisfied, false otherwise.
+     */
+    template <size_t Group = 0, std::enable_if_t<(Group < m_groups) && (Group >= 0), bool> = true>
+    bool check_constraints_impl() const
+    {
+        using LctState = typename std::tuple_element_t<Group, tupleLctStates>;
+
+        if (LctState::template get_num_subcompartments<InfectionState::Susceptible>() != 1) {
+            log_warning(
+                "Constraint check: The number of subcompartments for Susceptibles of group {} is should be one!",
```

Comment:
```suggestion
                "Constraint check: The number of subcompartments for Susceptibles of group {} should be one!",
```
Also for R and D below.